src/kubernetes.cc

```diff
@@ -961,6 +961,28 @@ json::value KubernetesReader::FindTopLevelOwner(
   return FindTopLevelOwner(ns, GetOwner(ns, ref->As<json::Object>()));
 }
 
+bool KubernetesReader::ValidateConfiguration() const {
+  try {
+    (void) QueryMaster(std::string(kKubernetesEndpointPath) + "/nodes?limit=1");
+  } catch (const QueryException& e) {
+    // Already logged.
+    return false;
+  }
+
+  if (CurrentNode().empty()) {
+    return false;
+  }
+
+  try {
+    (void) QueryMaster(std::string(kKubernetesEndpointPath) + "/pods?limit=1");
```

Comment:
For the future: by now this check is probably redundant, as `CurrentNode()` invokes this endpoint too. But we can fix that later.

Reply:
On second thought, since you're waiting for a response from @supriyagarg anyway, can you just move this above the `CurrentNode()` check?

Reply:
Done.